aws/product/awsec2product.go

```diff
@@ -0,0 +1,179 @@
+	dbAwsRegion, err := models.AwsRegionByPretty(sqlTx, region)
+	if err != nil && err.Error() == "sql: no rows in result set" {
+		var newDbAwsRegion models.AwsRegion
+		newDbAwsRegion.Pretty = region
+		newDbAwsRegion.Region = region
+		if err := newDbAwsRegion.Insert(sqlTx); err != nil {
+			logger.Error("Error during dbAwsRegion inserting", err.Error())
+			return err
+		}
+		dbAwsProduct.RegionID = newDbAwsRegion.ID
+	} else if err == nil {
+		dbAwsProduct.RegionID = dbAwsRegion.ID
+	} else {
+		logger.Error("Error during dbAwsRegion fetching", err.Error())
+		return err
+	}
+	return nil
+}
+
+// storeAttributes stores all the attributes from PricingAttribute
+// to models.AwsProductEc2
+func storeAttributes(attributes *PricingAttribute, dbAwsProduct *models.AwsProductEc2, logger jsonlog.Logger, sqlTx models.XODB) (err error) {
+	dbAwsProduct.InstanceType = attributes.InstanceType
+	if attributes.CurrentGeneration == "Yes" {
+		dbAwsProduct.CurrentGeneration = 1
+	}
+	if val, err := strconv.Atoi(attributes.Vcpu); err == nil {
+		dbAwsProduct.Vcpu = val
+	}
+	dbAwsProduct.Memory = attributes.Memory
+	dbAwsProduct.Storage = attributes.Storage
+	dbAwsProduct.NetworkPerformance = attributes.NetworkPerformance
+	dbAwsProduct.Tenancy = attributes.Tenancy
+	dbAwsProduct.OperatingSystem = attributes.OperatingSystem
+	if val, err := strconv.Atoi(attributes.Ecu); err == nil {
+		dbAwsProduct.Ecu = val
+	}
+	if attributes.LocationType == "AWS Region" {
+		err = storeRegionID(attributes.Location, dbAwsProduct, logger, sqlTx)
+	}
+	return err
+}
+
+// importResult parses the body returned by downloadJSON and
+// inserts the pricing in the database.
+func importResult(reader io.ReadCloser, logger jsonlog.Logger, sqlTx models.XODB) error {
+	defer reader.Close()
+
+	decoder := json.NewDecoder(reader)
+	var bodyMap EC2Pricing
+	if err := decoder.Decode(&bodyMap); err != nil {
+		return err
+	}
+	for _, product := range bodyMap.Products {
+		if product.ProductFamily == "Compute Instance" {
+			var dbAwsProduct models.AwsProductEc2
+
+			dbAwsProduct.Sku = string(product.Sku)
+			if err := storeAttributes(&product.Attributes, &dbAwsProduct, logger, sqlTx); err != nil {
+				return err
+			}
+			if err := dbAwsProduct.Insert(sqlTx); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func saveLastFetch(lastFetch *models.AwsFetchPricing, newEtag string, sqlTx models.XODB) error {
+	if lastFetch != nil {
+		lastFetch.Delete(sqlTx)
+	}
+	dbAfp := models.AwsFetchPricing{
+		Product: "ec2",
+		Etag:    newEtag,
+	}
+	return dbAfp.Save(sqlTx)
+}
+
+// downloadJSON requests AWS' API and returns the body after
+// checking if there's already the same version in the database.
+func downloadJSON(logger jsonlog.Logger, sqlTx models.XODB) (io.ReadCloser, error) {
+	hc := http.Client{}
+	req, err := http.NewRequest("GET", config.UrlEC2Pricing, nil)
+	if err != nil {
+		return nil, err
+	}
+	lastFetch, err := models.AwsFetchPricingByProduct(db.Db, "ec2")
```

Comment:
Please extract `"ec2"` into a constant. This is generally good practice, because a typo in a string can be hard to debug, while a typo in an identifier usually immediately causes a compile error.